src/test/scala/BatchWriterClientSpec.scala

```diff
@@ -47,92 +57,60 @@ class BatchWriterClientSpec extends ClientSpecification {
         "keen.queue.batch.timeout" -> "5 seconds",
         "keen.queue.max-events-per-collection" -> 250,
         "keen.queue.send-interval.events" -> 100,
-        "keen.queue.send-interval.duration" -> "2 seconds",
+        "keen.queue.send-interval.duration" -> "3600 seconds",
         "keen.queue.shutdown-delay" -> "0s"
       )
     ).withFallback(dummyConfig)
 
+
     "send queued events" in {
+      skipped("Requires refactoring because of random failures: https://github.com/keenlabs/KeenClient-Scala/issues/51")
+
       testEvents = generateTestEvents(5)
-      testEvents foreach (queueForTestCollection)
+      testEvents foreach queueForTestCollection
 
       // verify that the expected number of events are in the store
-      handleMap = store.getHandles(projectId)
-      handleMap.size must beEqualTo(1)
-      handleMap(collection).size must beEqualTo(5)
+      assertEventualStoreHandlesSize(1)
+      assertEventualStoreHandlesCollectionSize(5)
 
       // send the queued events
       client.sendQueuedEvents()
 
       // validate that the store is now empty
-      handleMap = store.getHandles(projectId)
-      handleMap.size must beEqualTo(0)
+      assertEventualStoreHandlesSize(0)
 
       // try sending events again, nothing should happen because the queue is empty
       client.sendQueuedEvents()
 
       // the store should still be empty
-      handleMap = store.getHandles(projectId)
-      handleMap.size must beEqualTo(0)
+      assertEventualStoreHandlesSize(0)
     }
 
-    "automatically send queued events when queue reaches keen.queue.send-interval.events" in {
-      testEvents = generateTestEvents(100)
+    "send queued events asynchronously when queue exceeds its send-interval.events limit" in {
+      skipped("Requires refactoring because of random failures: https://github.com/keenlabs/KeenClient-Scala/issues/51")
 
-      // queue the first 50 events
-      testEvents take 50 foreach (queueForTestCollection)
+      testEvents = generateTestEvents(queueConfig.getInt("keen.queue.send-interval.events"))
+      testEvents foreach queueForTestCollection
 
-      // verify that the expected number of events are in the store
-      handleMap = store.getHandles(projectId)
-      handleMap.size must beEqualTo(1)
-      handleMap(collection).size must beEqualTo(50)
-
-      // add the final 50 events
-      testEvents drop 50 foreach (queueForTestCollection)
-
-      // validate that the store is now empty as a result of sendQueuedEvents being automatically
-      // triggered with the queueing of the 100th event
-      Thread.sleep(300.millis.toMillis) // flush is async, wait for a beat
-      handleMap = store.getHandles(projectId)
-      handleMap.size must beEqualTo(0)
-    }
-
-    "automatically send queued events every keen.queue.send-interval.duration" in {
```

Comment:
Do not remove this test-case please, it looks like being independent of the others. Let's leave it but skip the test for now as you did with others.

Reply:
this one and one above verified implementation details instead of system behaviour and I removed then on purpose. as a replacement I suggested "send queued events asynchronously when queue exceeds its send-interval.events limit" but in order to make it pass consistently we need to rewrite some parts of production code to make it testable